File: C_Plus_Plus/1001-2000/p2095.cpp

using namespace std;

 struct ListNode {
      int val;
      ListNode *next;
      ListNode() : val(0), next(nullptr) {}
      ListNode(int x) : val(x), next(nullptr) {}
      ListNode(int x, ListNode *next) : val(x), next(next) {}
  };
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        if(head->next == nullptr)return nullptr;
        ListNode *prev = nullptr, *curr = head, *fast = head;
        while(fast && fast->next){
            prev = curr;
            curr = curr->next;
            fast = fast->next->next;
        }

        prev->next = curr->next;
        return head;
    }
};

File: C_Sharp/1001-2000/p2095.cs
    namespace p2095 {
    public class ListNode {
        public int val;
        public ListNode next;
        public ListNode(int val=0, ListNode next=null) {
            this.val = val;
            this.next = next;
        }
    }
    public class Solution {

        public ListNode DeleteMiddle(ListNode head) {
            if(head.next == null)return null;
            ListNode prev = null, curr = head, fast = head;
            while(fast != null && fast.next != null){
                prev = curr;
                curr = curr.next;
                fast = fast.next.next;
            }
            prev.next = curr.next;
            return head;
        }
    }
}